### ISSUE TITLE
BigQuery: SQLFluff treats ARRAY function as a keyword

### ISSUE THREAD

**Issue by sowusu-ba** [sqlfluff/sqlfluff#4790]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In my SQL,

```sql
SELECT
    ARRAY(SELECT x.users FROM UNNEST(array_one) AS x) AS users
FROM table_a
```

### Expected Behaviour

I expected SQLFluff to lint without any issues because ARRAY is also function in BIGQUERY

### Observed Behaviour

<img width="832" alt="image" src="https://user-images.githubusercontent.com/99834914/233137750-10f6f527-3b53-48ac-85e5-b74530f8d280.png">

I kept getting the error in the image above. This error however disappears when I set this in my config file

### How to reproduce

Run sqlfluff lint on the code with the config below

### Dialect

bigquery

### Version

2.0.5

### Configuration

```
[sqlfluff]
dialect = bigquery
templater = dbt
exclude_rules = AM04, AM07, CV01, CV02, CV04, CV08, CV10, LT05, LT12, RF05, ST01, ST02, ST06, ST07
max_line_length = 500

[sqlfluff:indentation]
## See https://docs.sqlfluff.com/en/stable/indentation.html
indented_joins = False
indented_using_on = False
indented_on_contents = True
template_blocks_indent = True
tab_space_size = 4
max_line_length = 500
indent_unit = space
allow_implicit_indents = True

#
[sqlfluff:templater]
unwrap_wrapped_queries = False
#
[sqlfluff:templater:jinja]
apply_dbt_builtins = True

# config settings for running with dbt templater
[sqlfluff:templater:dbt]
project_dir = ../..
profile=default


## Some rules can be configured directly from the config common to other rules
[sqlfluff:layout:type:set_operator]
line_position = touch

# [sqlfluff:layout:type:start_bracket]
# spacing_before = touch

# [sqlfluff:rules:layout.select_targets]
# wildcard_policy = multiple

#
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

#
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

#
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

#
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

#
[sqlfluff:rules:capitalisation.types]
# Data Types
extended_capitalisation_policy = consistent

#
[sqlfluff:rules:ambiguous.column_references]
# GROUP BY/ORDER BY column references
group_by_and_order_by_style = explicit


```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
